With new JNI, failed other compilation errors https://github.com/NVIDIA/spark-rapids/actions/runs/24177265116/job/70586032608?pr=14572

for spark 33x
`value setRowIndexOffset is not a member of org.apache.parquet.hadoop.metadata.BlockMetaData`

```
[INFO] --- scala:4.9.2:compile (scala-compile-first) @ rapids-4-spark-sql_2.12 ---
[INFO] Compiler bridge file: /home/runner/.m2/repository/.sbt/1.0/zinc/org.scala-sbt/org.scala-sbt-compiler-bridge_2.12-1.10.0-bin_2.12.21__61.0-1.10.0_20240505T232140.jar
[INFO] compiling 546 Scala sources and 36 Java sources to /home/runner/work/spark-rapids/spark-rapids/sql-plugin/target/spark331/classes ...
Error: ] /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/parquet/GpuParquetScan.scala:733: value setRowIndexOffset is not a member of org.apache.parquet.hadoop.metadata.BlockMetaData
Error: ] /home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/parquet/GpuParquetScan.scala:765: value setRowIndexOffset is not a member of org.apache.parquet.hadoop.metadata.BlockMetaData
Error: [ERROR] two errors found
```

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/parquet/GpuParquetScan.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/parquet/GpuParquetScan.scala
@@ -45,7 +45,7 @@ import com.nvidia.spark.rapids.jni.{DateTimeRebase, ParquetFooter, RmmSpark}
 import com.nvidia.spark.rapids.jni.fileio.{RapidsFileIO, RapidsInputFile, SeekableInputStream}
 import com.nvidia.spark.rapids.parquet.ParquetPartitionReader.{CopyRange, LocalCopy, PARQUET_MAGIC}
 import com.nvidia.spark.rapids.shims.{ColumnDefaultValuesShims, GpuParquetCrypto, GpuTypeShims, ShimFilePartitionReaderFactory, SparkShimImpl}
-import com.nvidia.spark.rapids.shims.parquet.{ParquetLegacyNanoAsLongShims, ParquetSchemaClipShims, ParquetStringPredShims}
+import com.nvidia.spark.rapids.shims.parquet.{GpuParquetUtilsShims, ParquetLegacyNanoAsLongShims, ParquetSchemaClipShims, ParquetStringPredShims}
 import org.apache.commons.io.output.{CountingOutputStream, NullOutputStream}
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
@@ -730,7 +730,7 @@ protected case class GpuParquetFileFilterHandler(
                   // Fix up the row index offset
                   val offsets = tableFooter.getRowIndexOffsets
                   if (offsets.nonEmpty) {
-                    block.setRowIndexOffset(offsets(0))
+                    GpuParquetUtilsShims.setRowIndexOffset(block, offsets(0))
                   }
                   val schema = new MessageType("root")
                   return ParquetFileInfoWithBlockMeta(filePath, Seq(block), file.partitionValues,
@@ -762,7 +762,7 @@ protected case class GpuParquetFileFilterHandler(
               s"Row index offsets length (${rowIndexOffsets.length}) != " +
               s"blocks count (${blocks.size()})")
             blocks.asScala.zip(rowIndexOffsets).foreach { case (block, offset) =>
-              block.setRowIndexOffset(offset)
+              GpuParquetUtilsShims.setRowIndexOffset(block, offset)
             }
             footer
           case _ =>

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/parquet/GpuParquetScan.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/parquet/GpuParquetScan.scala
@@ -760,7 +760,8 @@ protected case class GpuParquetFileFilterHandler(
             val blocks = footer.getBlocks
             require(blocks.size() == rowIndexOffsets.length,
               s"Row index offsets length (${rowIndexOffsets.length}) != " +
-              s"blocks count (${blocks.size()})")
+              s"blocks count (${blocks.size()}) for $filePath " +
+              s"[range=${file.start}:${file.length}]")
             blocks.asScala.zip(rowIndexOffsets).foreach { case (block, offset) =>
               GpuParquetUtilsShims.setRowIndexOffset(block, offset)
             }

--- a/sql-plugin/src/main/spark330/scala/com/nvidia/spark/rapids/shims/parquet/GpuParquetUtilsShims.scala
+++ b/sql-plugin/src/main/spark330/scala/com/nvidia/spark/rapids/shims/parquet/GpuParquetUtilsShims.scala
@@ -30,6 +30,11 @@ import org.apache.parquet.hadoop.metadata.{BlockMetaData, ColumnChunkMetaData}
 object GpuParquetUtilsShims {
 
   /**
+   * Sets the given row index offset on the BlockMetaData. No-op for Spark 3.3.x.
+   */
+  def setRowIndexOffset(block: BlockMetaData, offset: Long): Unit = {}
+
+  /**
    * Build a new BlockMetaData from an existing one, but with a new set of column chunks metadata.
    *
    * @param existingMetadata the existing BlockMetaData to copy row index offset from

--- a/sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shims/parquet/GpuParquetUtilsShims.scala
+++ b/sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shims/parquet/GpuParquetUtilsShims.scala
@@ -44,6 +44,13 @@ import org.apache.parquet.hadoop.metadata.{BlockMetaData, ColumnChunkMetaData}
 object GpuParquetUtilsShims {
 
   /**
+   * Sets the given row index offset on the BlockMetaData.
+   */
+  def setRowIndexOffset(block: BlockMetaData, offset: Long): Unit = {
+    block.setRowIndexOffset(offset)
+  }
+
+  /**
    * Build a new BlockMetaData from an existing one, but with a new set of column chunks metadata.
    *
    * @param existingMetadata the existing BlockMetaData to copy row count and row index offset from

